File: scripts/seed.py
#!/usr/bin/env python3
import urllib.request
import json
import sys
import datetime

BASE = 'http://localhost:3000/api'

def api_post(endpoint, data):
    req = urllib.request.Request(
        f'{BASE}{endpoint}',
        data=json.dumps(data).encode('utf-8'),
        headers={'Content-Type': 'application/json'},
        method='POST'
    )
    try:
        with urllib.request.urlopen(req) as resp:
            return json.loads(resp.read().decode('utf-8'))
    except Exception as e:
        print(f'ERROR POST {endpoint}: {e}')
        if hasattr(e, 'read'):
            print(e.read().decode('utf-8'))
        return None

def api_get(endpoint):
    req = urllib.request.Request(f'{BASE}{endpoint}')
    try:
        with urllib.request.urlopen(req) as resp:
            return json.loads(resp.read().decode('utf-8'))
    except Exception as e:
        print(f'ERROR GET {endpoint}: {e}')
        return None

# Check existing data
vehicles = api_get('/vehicles')
if vehicles and len(vehicles) > 0:
    print(f'Already have {len(vehicles)} vehicles, skipping seed')
    sys.exit(0)

# Create vehicles
v1 = api_post('/vehicles', {
    'name': 'Мой автомобиль',
    'brand': 'Toyota',
    'model': 'Camry',
    'year': 2020,
    'vin': '4T1BF1FK5EU123456',
    'licensePlate': 'А777МР77',
    'currentMileage': 85000,
    'color': 'Белый',
    'fuelType': 'petrol'
})
if not v1:
    print('Failed to create vehicle 1')
    sys.exit(1)
print(f'Created vehicle 1: {v1["id"]}')

v2 = api_post('/vehicles', {
    'name': 'Рабочая машина',
    'brand': 'Volkswagen',
    'model': 'Tiguan',
    'year': 2022,
    'vin': 'WVWZZZ5NZJW012345',
    'licensePlate': 'К888ОР99',
    'currentMileage': 42000,
    'color': 'Серый',
    'fuelType': 'diesel'
})
if not v2:
    print('Failed to create vehicle 2')
    sys.exit(1)
print(f'Created vehicle 2: {v2["id"]}')

# Create maintenance schedules
now = datetime.datetime.now()
next_month = (now.replace(day=1) + datetime.timedelta(days=32)).replace(day=15)
next_week = now + datetime.timedelta(days=5)
overdue = now - datetime.timedelta(days=30)

schedules_data = [
    {
        'vehicleId': v1['id'],
        'name': 'Замена масла',
        'description': 'Замена моторного масла и масляного фильтра',
        'intervalMileage': 10000,
        'intervalMonths': 12,
        'lastDate': (now - datetime.timedelta(days=365)).isoformat(),
        'lastMileage': 75000,
        'nextDate': next_week.isoformat(),
        'nextMileage': 85000,
        'isActive': True
    },
    {
        'vehicleId': v1['id'],
        'name': 'Замена тормозных колодок',
        'description': 'Проверка и замена передних/задних колодок',
        'intervalMileage': 30000,
        'intervalMonths': 24,
        'lastDate': (now - datetime.timedelta(days=400)).isoformat(),
        'lastMileage': 60000,
        'nextDate': overdue.isoformat(),
        'nextMileage': 90000,
        'isActive': True
    },
    {
        'vehicleId': v2['id'],
        'name': 'ТО-2',
        'description': 'Второе техническое обслуживание по регламенту',
        'intervalMileage': 15000,
        'intervalMonths': 12,
        'lastDate': (now - datetime.timedelta(days=180)).isoformat(),
        'lastMileage': 30000,
        'nextDate': next_month.isoformat(),
        'nextMileage': 45000,
        'isActive': True
    },
]

for s in schedules_data:
    result = api_post('/maintenance-schedules', s)
    if result:
        print(f'Created schedule: {result.get("name", "unknown")}')
    else:
        print(f'Failed to create schedule: {s["name"]}')

# Create maintenance records
records_data = [
    {'vehicleId': v1['id'], 'date': (now - datetime.timedelta(days=90)).isoformat(), 'mileage': 80000, 'cost': 5500, 'description': 'Замена масла Castrol 5W-40 и фильтра', 'workshop': 'Автосервис Мотор'},
    {'vehicleId': v1['id'], 'date': (now - datetime.timedelta(days=180)).isoformat(), 'mileage': 75000, 'cost': 12000, 'description': 'Замена передних колодок + диагностика', 'workshop': 'Официальный дилер Тойота'},
    {'vehicleId': v2['id'], 'date': (now - datetime.timedelta(days=30)).isoformat(), 'mileage': 40000, 'cost': 8500, 'description': 'Замена масла, фильтров, проверка ходовой', 'workshop': 'Автосервис Немец'},
    {'vehicleId': v2['id'], 'date': (now - datetime.timedelta(days=120)).isoformat(), 'mileage': 35000, 'cost': 3200, 'description': 'Замена салонного фильтра', 'workshop': 'Самостоятельно'},
    {'vehicleId': v1['id'], 'date': (now - datetime.timedelta(days=270)).isoformat(), 'mileage': 70000, 'cost': 15000, 'description': 'Замена ремня ГРМ и роликов', 'workshop': 'Автосервис Мотор'},
]

for r in records_data:
    result = api_post('/maintenance-records', r)
    if result:
        print(f'Created record: {result.get("id", "unknown")}')
    else:
        print(f'Failed to create record: {r["description"][:30]}')

# Create expenses
expenses_data = [
    {'vehicleId': v1['id'], 'category': 'fuel', 'amount': 3500, 'date': (now - datetime.timedelta(days=5)).isoformat(), 'description': 'АИ-95, 50 литров', 'supplier': 'Лукойл'},
    {'vehicleId': v1['id'], 'category': 'fuel', 'amount': 2800, 'date': (now - datetime.timedelta(days=40)).isoformat(), 'description': 'АИ-95, 40 литров', 'supplier': 'Газпромнефть'},
    {'vehicleId': v1['id'], 'category': 'wash', 'amount': 800, 'date': (now - datetime.timedelta(days=35)).isoformat(), 'description': 'Комплексная мойка', 'supplier': 'Мойка №1'},
    {'vehicleId': v1['id'], 'category': 'insurance', 'amount': 45000, 'date': (now - datetime.timedelta(days=130)).isoformat(), 'description': 'ОСАГО + КАСКО', 'supplier': 'Ингосстрах'},
    {'vehicleId': v1['id'], 'category': 'parts', 'amount': 6500, 'date': (now - datetime.timedelta(days=60)).isoformat(), 'description': 'Масло Castrol + фильтр', 'supplier': 'Exist.ru'},
    {'vehicleId': v1['id'], 'category': 'parking', 'amount': 1200, 'date': (now - datetime.timedelta(days=25)).isoformat(), 'description': 'Парковка ТЦ, месяц', 'supplier': 'ТЦ Мега'},
    {'vehicleId': v2['id'], 'category': 'fuel', 'amount': 4200, 'date': (now - datetime.timedelta(days=7)).isoformat(), 'description': 'ДТ, 60 литров', 'supplier': 'Лукойл'},
    {'vehicleId': v2['id'], 'category': 'fine', 'amount': 1500, 'date': (now - datetime.timedelta(days=65)).isoformat(), 'description': 'Превышение скорости', 'supplier': 'ГИБДД'},
    {'vehicleId': v2['id'], 'category': 'parts', 'amount': 9800, 'date': (now - datetime.timedelta(days=45)).isoformat(), 'description': 'Тормозные диски передние', 'supplier': 'AutoDoc'},
    {'vehicleId': v2['id'], 'category': 'wash', 'amount': 600, 'date': (now - datetime.timedelta(days=2)).isoformat(), 'description': 'Экспресс-мойка', 'supplier': 'Мойдодыр'},
]

for e in expenses_data:
    result = api_post('/expenses', e)
    if result:
        print(f'Created expense: {result.get("category", "unknown")} {result.get("amount", 0)}')
    else:
        print(f'Failed to create expense: {e["category"]} {e["amount"]}')

# Create parts
parts_data = [
    {'vehicleId': v1['id'], 'name': 'Масляный фильтр', 'partNumber': '90915-YZZN2', 'category': 'filter', 'cost': 450, 'purchaseDate': (now - datetime.timedelta(days=90)).isoformat(), 'supplier': 'Exist.ru', 'notes': 'Оригинал Toyota'},
    {'vehicleId': v1['id'], 'name': 'Моторное масло', 'partNumber': 'CASTROL-5W40-4L', 'category': 'oil', 'cost': 3200, 'purchaseDate': (now - datetime.timedelta(days=90)).isoformat(), 'supplier': 'Exist.ru', 'notes': 'Castrol EDGE 5W-40, 4 литра'},
    {'vehicleId': v2['id'], 'name': 'Салонный фильтр', 'partNumber': '5Q0-819-653', 'category': 'filter', 'cost': 1200, 'purchaseDate': (now - datetime.timedelta(days=120)).isoformat(), 'supplier': 'AutoDoc', 'notes': 'Угольный фильтр'},
]

for p in parts_data:
    result = api_post('/parts', p)
    if result:
        print(f'Created part: {result.get("name", "unknown")}')
    else:
        print(f'Failed to create part: {p["name"]}')

print('\nSeeding complete!')

File: prisma/seed.ts
import { PrismaClient } from '@prisma/client'

const prisma = new PrismaClient()

async function main() {
  // Clean up existing data
  await prisma.part.deleteMany()
  await prisma.expense.deleteMany()
  await prisma.maintenanceRecord.deleteMany()
  await prisma.maintenanceSchedule.deleteMany()
  await prisma.vehicle.deleteMany()

  // Create vehicles
  const vehicle1 = await prisma.vehicle.create({
    data: {
      name: 'Мой автомобиль',
      brand: 'Toyota',
      model: 'Camry',
      year: 2020,
      vin: '4T1BF1FK5EU123456',
      licensePlate: 'А 777 АА 77',
      currentMileage: 85000,
      color: 'Чёрный',
      fuelType: 'petrol',
    },
  })

  const vehicle2 = await prisma.vehicle.create({
    data: {
      name: 'Жена',
      brand: 'Kia',
      model: 'Rio',
      year: 2022,
      vin: 'XWEHN511AB023456',
      licensePlate: 'В 123 ОР 99',
      currentMileage: 32000,
      color: 'Белый',
      fuelType: 'petrol',
    },
  })

  // Create maintenance schedules
  const schedule1 = await prisma.maintenanceSchedule.create({
    data: {
      vehicleId: vehicle1.id,
      name: 'Замена масла',
      description: 'Замена моторного масла и фильтра',
      intervalMileage: 10000,
      intervalMonths: 12,
      lastDate: new Date('2025-08-15'),
      lastMileage: 75000,
      nextDate: new Date('2026-02-15'),
      nextMileage: 85000,
    },
  })

  const schedule2 = await prisma.maintenanceSchedule.create({
    data: {
      vehicleId: vehicle1.id,
      name: 'Замена тормозных колодок',
      description: 'Замена передних и задних колодок',
      intervalMileage: 30000,
      intervalMonths: 24,
      lastDate: new Date('2024-11-01'),
      lastMileage: 60000,
      nextDate: new Date('2026-05-01'),
      nextMileage: 90000,
    },
  })

  const schedule3 = await prisma.maintenanceSchedule.create({
    data: {
      vehicleId: vehicle2.id,
      name: 'Замена масла',
      description: 'Замена моторного масла и фильтра',
      intervalMileage: 15000,
      intervalMonths: 12,
      lastDate: new Date('2025-06-01'),
      lastMileage: 25000,
      nextDate: new Date('2026-03-01'),
      nextMileage: 40000,
    },
  })

  // Create maintenance records
  const now = new Date()
  const records = [
    {
      vehicleId: vehicle1.id,
      scheduleId: schedule1.id,
      date: new Date('2025-08-15'),
      mileage: 75000,
      cost: 5500,
      description: 'Замена масла Castrol 5W-40 и фильтра',
      workshop: 'Автосервис «Мотор»',
    },
    {
      vehicleId: vehicle1.id,
      scheduleId: null,
      date: new Date('2025-06-20'),
      mileage: 72000,
      cost: 12000,
      description: 'Замена свечей зажигания',
      workshop: 'Тойота Центр',
    },
    {
      vehicleId: vehicle1.id,
      scheduleId: schedule2.id,
      date: new Date('2024-11-01'),
      mileage: 60000,
      cost: 8500,
      description: 'Замена передних тормозных колодок',
      workshop: 'Автосервис «Мотор»',
    },
    {
      vehicleId: vehicle2.id,
      scheduleId: schedule3.id,
      date: new Date('2025-06-01'),
      mileage: 25000,
      cost: 4200,
      description: 'Замена масла Shell 5W-30',
      workshop: 'KIA Сервис',
    },
    {
      vehicleId: vehicle2.id,
      scheduleId: null,
      date: new Date('2025-04-10'),
      mileage: 20000,
      cost: 3200,
      description: 'Диагностика ходовой',
      workshop: 'KIA Сервис',
    },
  ]

  for (const record of records) {
    await prisma.maintenanceRecord.create({ data: record })
  }

  // Create expenses
  const expenses = [
    {
      vehicleId: vehicle1.id,
      category: 'fuel',
      amount: 3500,
      date: new Date(now.getTime() - 2 * 24 * 60 * 60 * 1000),
      description: 'АИ-95, 50 литров',
      supplier: 'Лукойл',
    },
    {
      vehicleId: vehicle1.id,
      category: 'wash',
      amount: 800,
      date: new Date(now.getTime() - 3 * 24 * 60 * 60 * 1000),
      description: 'Комплексная мойка',
      supplier: 'Мойка №1',
    },
    {
      vehicleId: vehicle1.id,
      category: 'insurance',
      amount: 25000,
      date: new Date(now.getTime() - 15 * 24 * 60 * 60 * 1000),
      description: 'ОСАГО на 1 год',
      supplier: 'Росгосстрах',
    },
    {
      vehicleId: vehicle1.id,
      category: 'parts',
      amount: 4500,
      date: new Date(now.getTime() - 10 * 24 * 60 * 60 * 1000),
      description: 'Воздушный фильтр',
      supplier: 'Автодок',
    },
    {
      vehicleId: vehicle1.id,
      category: 'fuel',
      amount: 3200,
      date: new Date(now.getTime() - 7 * 24 * 60 * 60 * 1000),
      description: 'АИ-95, 45 литров',
      supplier: 'Газпромнефть',
    },
    {
      vehicleId: vehicle1.id,
      category: 'parking',
      amount: 300,
      date: new Date(now.getTime() - 1 * 24 * 60 * 60 * 1000),
      description: 'Парковка ТЦ',
      supplier: '',
    },
    {
      vehicleId: vehicle2.id,
      category: 'fuel',
      amount: 2800,
      date: new Date(now.getTime() - 4 * 24 * 60 * 60 * 1000),
      description: 'АИ-92, 40 литров',
      supplier: 'Лукойл',
    },
    {
      vehicleId: vehicle2.id,
      category: 'fine',
      amount: 500,
      date: new Date(now.getTime() - 20 * 24 * 60 * 60 * 1000),
      description: 'Превышение скорости',
      supplier: 'ГИБДД',
    },
    {
      vehicleId: vehicle2.id,
      category: 'wash',
      amount: 600,
      date: new Date(now.getTime() - 5 * 24 * 60 * 60 * 1000),
      description: 'Экспресс-мойка',
      supplier: 'Мойка №3',
    },
    {
      vehicleId: vehicle1.id,
      category: 'other',
      amount: 1500,
      date: new Date(now.getTime() - 12 * 24 * 60 * 60 * 1000),
      description: 'Антидождь для стёкол',
      supplier: 'Автозапчасти',
    },
  ]

  for (const expense of expenses) {
    await prisma.expense.create({ data: expense })
  }

  // Create parts
  const parts = [
    {
      vehicleId: vehicle1.id,
      name: 'Масляный фильтр',
      partNumber: '90915-YZZN2',
      category: 'engine',
      cost: 650,
      purchaseDate: new Date('2025-08-10'),
      supplier: 'Автодок',
      notes: 'Оригинал Toyota',
    },
    {
      vehicleId: vehicle1.id,
      name: 'Передние тормозные колодки',
      partNumber: '04465-33471',
      category: 'brakes',
      cost: 3500,
      purchaseDate: new Date('2024-10-25'),
      supplier: 'Экзист',
      notes: 'Оригинал',
    },
    {
      vehicleId: vehicle2.id,
      name: 'Салонный фильтр',
      partNumber: '97133-H1000',
      category: 'interior',
      cost: 850,
      purchaseDate: new Date('2025-05-15'),
      supplier: 'Автодок',
      notes: 'Угольный фильтр',
    },
  ]

  for (const part of parts) {
    await prisma.part.create({ data: part })
  }

  console.log('Seed data created successfully!')
}

main()
  .catch((e) => {
    console.error(e)
    process.exit(1)
  })
  .finally(async () => {
    await prisma.$disconnect()
  })
